dbms/src/Flash/Mpp/MinTSOScheduler.cpp

```diff
@@ -393,4 +424,18 @@ bool MinTSOScheduler::updateMinQueryId(const MPPQueryId & query_id, const bool r
     return force_scheduling;
 }
 
+MinTSOScheduler::GroupEntry & MinTSOScheduler::getOrCreateGroupEntry(const String & resource_group_name)
+{
+    auto iter = scheduler_entries.find(resource_group_name);
+    if (iter == scheduler_entries.end())
+    {
+        GET_RESOURCE_GROUP_METRIC(tiflash_task_scheduler, type_thread_hard_limit, resource_group_name)
+            .Set(thread_hard_limit);
+        GET_RESOURCE_GROUP_METRIC(tiflash_task_scheduler, type_thread_soft_limit, resource_group_name)
+            .Set(thread_soft_limit);
+        iter = scheduler_entries.insert({resource_group_name, GroupEntry(resource_group_name)}).first;
+    }
```

Comment:
https://github.com/pingcap/tiflash/pull/8072/files#diff-d3878052df816dee12a6e833414384208e672da08ba05993037f6f16e8b53c84L76-L84
Maybe `GET_METIRCS(),set(0)` needs to be called?

Reply:
Calling once is ok, because `thread_soft_limit/thread_hard_limit` doesn't change in the process of running queries.

Reply:
What about `GET_RESOURCE_GROUP_METRIC(tiflash_task_scheduler, type_active_tasks_count).Set(0);`？

Reply:
No need to add `GET_RESOURCE_GROUP_METRIC(tiflash_task_scheduler, type_active_tasks_count).Set(0);`?
because this metric only meaningful when its resource group came to tiflash.

Reply:
done. add when create new group entry